src/material/datepicker/calendar-body.html

```diff
@@ -26,8 +26,10 @@
       [style.paddingBottom]="_cellPadding">
     {{_firstRowOffset >= labelMinRequiredCells ? label : ''}}
   </td>
-  <td *ngFor="let item of row; let colIndex = index"
-      role="gridcell"
+  <div role="gridcell" *ngFor="let item of row; let colIndex = index" style="display: contents;">
```

Comment:
This is unfortunately invalid HTML- a `<tr>` element cannot directly contain a `<div>`. 

We should be able to get something working with the `<button>` inside the `gridcell`- what did you run into with that approach?

Reply:
To start, it broke many tests because they needed to select the button inside the `.mat-calendar-body-cell` instead of the `.mat-calendar-body-cell` itself. Rather than running into specific problems, it seemed more like there could be an easier way to do this.

Wrapping the button with a `display: contents;` makes things easier because then the rest of the controller code and the styles only need to target the button, and do not need to be aware of the gridcell.

I can also try doing something like this, which should be valid html

```
<td role="gridcell" style="display: contents;">
  <div role="button" tabindex="0" class="mat-calendar-body-cell" ...>
    ...
  </div>
</td>
```

Reply:
I did run into styling problems by using the `button` tag instead of `div role="button"`. But I would expect the styling to be fixable by adding more CSS rules.